ccl/backupccl: TestClusterRestoreFailCleanup failed

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7800462?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7800462?buildTab=artifacts#/) on release-22.2 @ [b1378718eb74f0412035acee659ce3b7941ab3dd](https://github.com/cockroachdb/cockroach/commits/b1378718eb74f0412035acee659ce3b7941ab3dd):


```
=== RUN   TestClusterRestoreFailCleanup
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestClusterRestoreFailCleanup3225252859
    test_log_scope.go:79: use -show-logs to present logs inline
    full_cluster_backup_restore_test.go:607: error executing 'BACKUP TO 'nodelocal://1/missing-ssts'': pq: failed to run backup: writing forward-compat metadata sst: pebble: keys must be added in strictly increasing order: "stats/\x8c\x88"/0,0#0,SET, "stats/\x8c\x88"/0,0#0,SET
    panic.go:522: -- test log scope end --
/artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/TestDataDriven_restore-validation-only3829916060/full_database_backup/2022/12/02-083744.09/data/818871875263234049.ssttest logs left over in: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestClusterRestoreFailCleanup3225252859
--- FAIL: TestClusterRestoreFailCleanup (218.04s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7895540?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7895540?buildTab=artifacts#/) on release-22.2 @ [45cceae6a1277ab1970d24806cb57230814bf494](https://github.com/cockroachdb/cockroach/commits/45cceae6a1277ab1970d24806cb57230814bf494):


```
=== RUN   TestClusterRestoreFailCleanup
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestClusterRestoreFailCleanup2591473907
    test_log_scope.go:79: use -show-logs to present logs inline
    full_cluster_backup_restore_test.go:607: error executing 'BACKUP TO 'nodelocal://1/missing-ssts'': pq: failed to run backup: writing forward-compat metadata sst: pebble: keys must be added in strictly increasing order: "stats/\x8c\x88"/0,0#0,SET, "stats/\x8c\x88"/0,0#0,SET
    panic.go:522: -- test log scope end --
/artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/TestDataDriven_restore-validation-only1264175192/full_database_backup/2022/12/09-074312.34/data/820842963104563201.ssttest logs left over in: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestClusterRestoreFailCleanup2591473907
--- FAIL: TestClusterRestoreFailCleanup (231.58s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/backupinfo/BUILD.bazel
+++ b/pkg/ccl/backupccl/backupinfo/BUILD.bazel
@@ -37,7 +37,6 @@ go_library(
         "//pkg/sql/sem/tree",
         "//pkg/sql/stats",
         "//pkg/storage",
-        "//pkg/util",
         "//pkg/util/ctxgroup",
         "//pkg/util/encoding",
         "//pkg/util/hlc",

--- a/pkg/ccl/backupccl/backupinfo/manifest_handling.go
+++ b/pkg/ccl/backupccl/backupinfo/manifest_handling.go
@@ -44,7 +44,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/stats"
-	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
@@ -92,7 +91,7 @@ var WriteMetadataSST = settings.RegisterBoolSetting(
 	settings.TenantWritable,
 	"kv.bulkio.write_metadata_sst.enabled",
 	"write experimental new format BACKUP metadata file",
-	util.ConstantWithMetamorphicTestBool("write-metadata-sst", false),
+	false,
 )
 
 // IsGZipped detects whether the given bytes represent GZipped data. This check

